src/test/java/echoserver/FakeServerSocket.java

```diff
@@ -8,15 +8,16 @@
 
 class FakeServerSocket extends ServerSocket {
     private List<FakeClientSocket> fakeClientSockets = new ArrayList<>();
-    private final OutputStream output;
+    private List<FakeClientSocket> acceptedClientSockets = new ArrayList<>();
 
-    FakeServerSocket(OutputStream output) throws IOException {
-        this.output = output;
+    public FakeServerSocket() throws IOException {
     }
 
     @Override
     public Socket accept() {
-        return fakeClientSockets.remove(0);
+        var socket = fakeClientSockets.remove(0);
+        acceptedClientSockets.add(socket);
+        return socket;
     }
 
     void setupFakeClientSockets() {
```

Comment:
Think it would help to extract a method that sets up a single fakeclientsocket.

Reply:
Good point, I'll add this. Thanks